From 295e3f9dfceabce99d72af6b0c64e64310433c46 Mon Sep 17 00:00:00 2001
From: Jeremy Gebben <jgebben@codeaurora.org>
Date: Tue, 26 Apr 2011 15:54:03 -0600
Subject: [PATCH] msm: kgsl: use relaxed readl and writel for memory
 read/write

Calling code should have barriers where necessary rather than
relying on barriers in readl() and writel().

Change-Id: Icfd50ce59a8ea5412f8e95985c27d1d3572de4e6
Signed-off-by: Jeremy Gebben <jgebben@codeaurora.org>
---
 drivers/gpu/msm/kgsl_sharedmem.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/msm/kgsl_sharedmem.c b/drivers/gpu/msm/kgsl_sharedmem.c
index 0d4a0e7..f21b889 100644
--- a/drivers/gpu/msm/kgsl_sharedmem.c
+++ b/drivers/gpu/msm/kgsl_sharedmem.c
@@ -616,7 +616,7 @@ void kgsl_sharedmem_free(struct kgsl_memdesc *memdesc)
 	if (offsetbytes + sizeof(unsigned int) > memdesc->size)
 		return -ERANGE;
 
-	*dst = readl(memdesc->hostptr + offsetbytes);
+	*dst = readl_relaxed(memdesc->hostptr + offsetbytes);
 	return 0;
 }
 EXPORT_SYMBOL(kgsl_sharedmem_readl);
@@ -631,7 +631,7 @@ void kgsl_sharedmem_free(struct kgsl_memdesc *memdesc)
 
 	kgsl_cffdump_setmem(memdesc->physaddr + offsetbytes,
 		src, sizeof(uint));
-	writel(src, memdesc->hostptr + offsetbytes);
+	writel_relaxed(src, memdesc->hostptr + offsetbytes);
 	return 0;
 }
 EXPORT_SYMBOL(kgsl_sharedmem_writel);
-- 
1.7.5.4

